### ISSUE TITLE
MIGHT BE SEVERE BUG: abnormal data points may get when do EOS test (Useful to know one needs to use consistent kpoint meshes in one EOS test.)

### ISSUE THREAD

**Issue by kirk0830** [deepmodeling/abacus-develop#4011]
### Details
Firstly I post INPUT contents here:
```plaintext
INPUT_PARAMETERS
calculation relax # test; scf; relax; nscf; get_wf; get_pchg
ecutwfc 70.0 # #energy cutoff for wave functions
basis_type pw # PW; LCAO in pw; LCAO
symmetry 0 # the control of symmetry
pseudo_dir                      . # / #the directory containing pseudo files
scf_nmax 500 # #number of electron iterations
scf_thr                         1e-08  # charge density error
dft_functional PBE # exchange correlation functional
cal_force 1 # if calculate the force at the end of the electronic iteration
cal_stress 1 # calculate the stress or not
nspin 1 # 1: single spin; 2: up and down spin; 4: noncollinear spin
```
It is quite specific, not always seen:
for Al, FCC seems good:
![eos_Al_FCC](https://github.com/deepmodeling/abacus-develop/assets/67682086/02b38b8c-5196-4025-83f9-eaf3d13f42e3)
But something seems wrong for BCC:
![eos_Al_BCC](https://github.com/deepmodeling/abacus-develop/assets/67682086/041c91fc-8b18-45cd-972e-6da912d5e163)
for Li, **on the contrary**, BCC seems good:
![eos_Li_BCC](https://github.com/deepmodeling/abacus-develop/assets/67682086/39145a8c-a82a-41be-bbe0-742bd0e92a36)
FCC seems good too:
![eos_Li_FCC](https://github.com/deepmodeling/abacus-develop/assets/67682086/7466a65e-c20a-440b-a151-1c13ee96b545)
but Diamond structure seems terrible:
![eos_Li_DIAMOND](https://github.com/deepmodeling/abacus-develop/assets/67682086/67c9636a-cfe5-4658-94d8-70bb75c3d130)

in summary in my test suite, the EOS cases are like:
(cross represents abnormal case)   
|element|BCC|FCC|Diamond|
|--|--|--|--|
|Li|√|√|×|
|Be|√|×|√|
|B|√|√|√|
|C|×|√|√|
|Na|×|√|√|
|Mg|√|√|√|
|Al|×|√|√|
|Si|√|×|√|
|S|×|√|√|
|K|√|√|√|
|Ca|√|√|√|
|Sc|√|√|√|
|Ti|√|√|√|
|V|√|√|√|
|Cr|?|?|?|
|Mn|√|√|√|
|Fe|√|√|√|
|Co|√|√|√|
|Ni|√|√|√|
|Cu|×|√|√|
|Zn|√|×|×|
|Ga|√|√|×|

The test cases of Cr is special:
![eos_Cr_BCC](https://github.com/deepmodeling/abacus-develop/assets/67682086/4afc287a-8d3f-4d41-b494-f23fc58dff62)
![eos_Cr_FCC](https://github.com/deepmodeling/abacus-develop/assets/67682086/7b02c8d7-7763-4416-849a-e6dcce98b797)
![eos_Cr_DIAMOND](https://github.com/deepmodeling/abacus-develop/assets/67682086/074553a3-a734-4703-b770-5a8c53119065)

### Have you read FAQ on the online manual http://abacus.deepmodeling.com/en/latest/community/faq.html

- [X] Yes, I have read the FAQ part on online manual.

### Task list for Issue attackers (only for developers)

- [ ] Understand the problem or question described by the user.
- [ ] Check if the issue is a known problem or has been addressed in the documentation.
- [ ] Test the issue or problem on a similar system or environment, if possible.
- [ ] Identify the root cause or provide clarification on the user's question.
- [ ] Provide a step-by-step guide, including any necessary resources, to resolve the issue or answer the question.
- [ ] If the issue is related to documentation, update the documentation to prevent future confusion (optional).
- [ ] If the issue is related to code, consider implementing a fix or improvement (optional).
- [ ] Review and incorporate any relevant feedback from users or developers.
- [ ] Ensure the user's issue is resolved or their question is answered and close the ticket.

**Comment by kirk0830** [comment]
# Compare with other DFT codes
## Cases of abnormal EOS profiles which are not parabolic:
For Al:
![image](https://github.com/deepmodeling/abacus-develop/assets/67682086/321b3e61-87cd-45a8-aff2-42068cd04aff)

For Li:
![image](https://github.com/deepmodeling/abacus-develop/assets/67682086/29bbb86f-f49c-4c77-9787-0939e22ccee5)

## Cases of Cr in which PseudoDojo v0.4 works bad
**One may argue that for Cr case, it may result from the parameterization of PseudoDojo v0.4 pseudopotential**, but if compare with ABINIT/PseudoDojo-v0.5 (sr), SIESTA/PseudoDojo-v0.4 (sr)：
![image](https://github.com/deepmodeling/abacus-develop/assets/67682086/ee0204ca-5792-4a1f-9082-aae74b53b132)
Seems PseudoDojo v0.4 works well on SIESTA, PseudoDojo v0.5 works well on ABINIT. Therefore I doubt whether there are any mistakes of implementation.

**Comment by kirk0830** [comment]
APNS EOS test will pause till this bug is fixed.
Batch of inputs which will yield questionable data:
[APNS_EOS_testsuite.zip](https://github.com/deepmodeling/abacus-develop/files/15019289/APNS_EOS_testsuite.zip)

**Comment by kirk0830** [comment]
this problem may result from inconsistency and unconvergence of kpoints, will use consistent kpoint meshes in one EOS test.

**Comment by kirk0830** [comment]
I noticed that all-electron data reference uses a quite dense kpoint mesh, kspacing is fixed at 0.06 Angstrom-1, then keep unchanged for differently scaled volume. I will update the APNS workflow

**Comment by kirk0830** [comment]
So if one wants to perform EOS test, remember:
- use a dense kmesh usually helps if want to directly get physical information
- **use fixed kmesh on all volumes**, to avoid the not-converged-kmesh caused incontinouity of EOS line (even small, will matter if calculate delta value)
